### DISCUSSION TITLE
Pump/display/app state sync

### DISCUSSION THREAD

**Discussion by njsaunders** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#786] [Q&A] [answered]
Hi - 

First of all, thank you so much @visualapproach and contributors for all the effort that's gone into this - Seriously impressive!

I've just finished my build for my 54154 unit, got the ESP on the network, and can turn the pump/heater on and off. There's a few things that aren't working as expected, and just wondered if these were as expected/bad assumptions on my part.

Observations:

1) The buttons on my pump don't reflect the state of the configuration set in the app and vice versa - I had "assumed" that this would be a passthrough and that pump settings (Bubbles on etc) would be reflected in the app, and via versa. Right now pump and heater are on controlled via the app, but the LEDs on the pump aren't illuminated.

2) Same with Target Target temperature - The temp set on the unit isn't reflected in app,  and the temp set in app isn't reflected on unit.

Appreciate any advice anyone could offer!

Thanks,

Neil

**Comment by visualapproach** [comment, marked as answer]
Hi! Thx, it works as expected. 4w pumps is designed so it's not possible to know what the display is showing and it doesn't know what the ESP is doing.
6 wire is better in that regard.

**Reply by njsaunders** [reply]
Thought that would be the case - Thanks.

Have sent some beer money as a thank you!

Cheers,

Neil

**Reply by njsaunders** [reply]
Oh and I was also going to raise a PR for the documentation but couldn't see the source. Confirmed working with "Lay-Z-Spa BW14294 Saint Tropez" (54154), 4 Wire, Egg Shape, V2B PCB & Red LLC (2).

**Reply by visualapproach** [reply]
Thank you @njsaunders really appreciate it!

I'll update the manual with your info. It's a word document that's only exported as pdf to github so it's not really editable for people unfortunately.
